esphome/components/lvgl/btnmatrix.py

```diff
@@ -0,0 +1,275 @@
+from esphome import automation
+import esphome.codegen as cg
+from esphome.components.key_provider import KeyProvider
+import esphome.config_validation as cv
+from esphome.const import CONF_ID, CONF_WIDTH
+from esphome.cpp_generator import MockObj
+
+from .automation import action_to_code
+from .btn import lv_btn_t
+from .defines import (
+    BTNMATRIX_CTRLS,
+    CONF_BUTTONS,
+    CONF_CONTROL,
+    CONF_ITEMS,
+    CONF_KEY_CODE,
+    CONF_MAIN,
+    CONF_ONE_CHECKED,
+    CONF_ROWS,
+    CONF_SELECTED,
+    CONF_TEXT,
+)
+from .helpers import lvgl_components_required
+from .lv_validation import key_code, lv_bool
+from .lvcode import lv, lv_add, lv_expr
+from .schemas import automation_schema
+from .types import (
+    LV_BTNMATRIX_CTRL,
+    LV_STATE,
+    LvBoolean,
+    LvCompound,
+    LvType,
+    ObjUpdateAction,
+    char_ptr,
+    lv_pseudo_button_t,
+)
+from .widget import Widget, WidgetType, get_widgets, widget_map
+
+CONF_BUTTONMATRIX = "buttonmatrix"
+CONF_BUTTONMATRIX_TEXT_LIST_ID = "buttonmatrix_text_list_id"
+
+LvBtnmBtn = LvBoolean(
+    str(cg.uint16),
+    parents=(lv_pseudo_button_t,),
+)
+BTNM_BTN_SCHEMA = cv.Schema(
```

Comment:
It would also be more preferred to not shorten everything down. It is better to be clear about what this is exactly.

> It’s preferred to use long variable/function names over short and non-descriptive ones.
https://esphome.io/guides/contributing#codebase-standards
```suggestion
BUTTON_MATRIX_BUTTON_SCHEMA = cv.Schema(
```